pkg/kubectl/cmd/drain.go

```diff
@@ -385,35 +418,57 @@ func (o *DrainOptions) getPodsForDeletion() (pods []api.Pod, err error) {
 	return pods, nil
 }
 
-// deletePods deletes the pods on the api server
-func (o *DrainOptions) deletePods(pods []api.Pod) error {
-	deleteOptions := api.DeleteOptions{}
+func (o *DrainOptions) evictPod(pod api.Pod) error {
+	deleteOptions := &api.DeleteOptions{}
 	if o.GracePeriodSeconds >= 0 {
 		gracePeriodSeconds := int64(o.GracePeriodSeconds)
 		deleteOptions.GracePeriodSeconds = &gracePeriodSeconds
 	}
+	eviction := &policy.Eviction{
+		TypeMeta: unversioned.TypeMeta{
+			APIVersion: "policy/v1alpha1",
+			Kind:       "Eviction",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name:      pod.Name,
+			Namespace: pod.Namespace,
+		},
+		DeleteOptions: deleteOptions,
+	}
+	return o.client.Policy().Evictions(eviction.Namespace).Evict(eviction)
+}
 
+// deletePods deletes the pods on the api server
+func (o *DrainOptions) deletePods(pods []api.Pod) error {
+	if len(pods) == 0 {
+		return nil
+	}
+	ifTooManyRequests := false
+	var e error
 	for _, pod := range pods {
-		err := o.client.Core().Pods(pod.Namespace).Delete(pod.Name, &deleteOptions)
+		err := o.evictPod(pod)
+		if apierrors.IsTooManyRequests(err) {
+			fmt.Println("IsTooManyRequests", err)
+			e = err
+			ifTooManyRequests = true
```

Comment:
I'm following the [naming convention](https://github.com/kubernetes/kubernetes/pull/35483/files#diff-d8c7f981f26878f354974019a9484816R412).